### ISSUE TITLE
import/order ignores pathGroups order

### ISSUE THREAD

**Issue by grechdev** [benmosher/eslint-plugin-import#1982]
eslint ignores the order in which imports are listed

example imports:
```
import React from 'react';

import Test from "components/Test";  
import Api from "pages/api";
```

eslint must throw an error saying that import from "pages/api" must be over "components/Test" and also be separated with a new line but it ignores this

eslint-plugin-import and eslint-import-resolver-typescript are already installed so I'm sure that I made a mistake somewhere, but where?

eslint rule:
```"import/order": ["error", {
      "newlines-between": "always-and-inside-groups",
      "groups": ["builtin", "external", "internal"],
      "pathGroups": [  
        {  
          "pattern": "react",  
          "group": "external",  
          "position": "before"  
        },  
        {  
          "pattern": "~/**",  
          "group": "external",  
          "position": "before"  
        },  
        {  
          "pattern": "~/pages/**",  
          "group": "internal"  
        },  
        {  
          "pattern": "~/components/**",  
          "group": "internal"  
        },  
        {  
          "pattern": "~/startup/**",  
          "group": "internal"  
        },  
        {  
          "pattern": "~/hooks/**",  
          "group": "internal"  
        },  
        {  
          "pattern": "~/helpers/**",  
          "group": "internal"  
        },  
        {  
          "pattern": "~/generated/**",  
          "group": "internal"  
        },  
        {  
          "pattern": "~/styles/**",  
          "group": "internal"  
        }  
      ],  
      "pathGroupsExcludedImportTypes": ["react"],  
      "alphabetize": {  
        "order": "asc",  
        "caseInsensitive": true  
      }  
    }]
```
Project structure:  
![image](https://user-images.githubusercontent.com/54347665/106421578-051e8b00-646e-11eb-8905-978d06e8a233.png)

**Comment by ljharb** [comment]
Your config has `~/pages` but your code example just has `pages/`.

Can you make a repro repo?

**Comment by ludwig-pro** [comment]
I think there is a bug when ESLint load the config, so config-validator.js : 

```sh
Configuration for rule "import/order" is invalid:
	Value {"newlines-between":"always","groups":[["builtin","external"],["internal","index","sibling","parent"]],"alphabetize":{"order":"asc","caseInsensitive":true},"pathGroups":[{"pattern":"~/**","group":"internal"}],"pathGroupsExcludedImportTypes":[]} should NOT have additional properties.
```

**Comment by ludwig-pro** [comment]
maybe related to https://github.com/benmosher/eslint-plugin-import/issues/1682

**Comment by yuri-sakharov** [comment]
Doesn't work `.scss` order 
```
"import/order": ["error", {
            "groups": ["external", ["internal", "parent", "sibling", "index"], "unknown"],
            "pathGroups": [
                {
                    "pattern": "*.scss",
                    "patternOptions": { "matchBase": true },
                    "group": "unknown",
                    "position": "after"
                }
            ],
            "pathGroupsExcludedImportTypes": [],
            "newlines-between": "always"
        }],
```

result is still same 
```
import './AvatarsGroup.scss';
import { Avatar } from 'components/Avatar';
import { PlusIndicator } from 'components/PlusIndicator';
```
Any ideas?

**Comment by ludwig-pro** [comment]
look at the ESLint output in vscode if there is any mistake with your config. 

for me, autofix works again after cleaning extends / plugins (there are conflits somethings)

**Comment by yuri-sakharov** [comment]
@ludwig-pro  thank you for response and looks like there is issues with it #1239

**Comment by nickvirdenlegalshield** [comment]
@yuri-sakharov I was having trouble just like you with other solutions, such as with solutions by @constgen and @MoradiDavijani from https://github.com/import-js/eslint-plugin-import/issues/1239, which were helpful in solving the issue.

After some toying around with it, I figured out that I had to set `warnOnUnassignedImports` to `true`.

Once I did that, it started picking up on the unassigned imports (e.g. `import './foo.scss';`, but it would put them after `sibling` instead at the very end of the import order like I wanted - after `object`.

Here's my configuration:

**Eslint version:** 7.31.0
**eslint-plugin-import version:** 2.24.0

### Import order
```jsx
import './foo.scss';
import barModule from '../../barModule';
import bazModule from './bazModule';
```

### ESLint configuration
```js
'import/order': [
    // other config here
    'pathGroups': [
      {
        pattern: '{.,..}/*\.scss', // same directory only
        // pattern: '{.,..}/**/*\.scss' // same & outside directories (e.g. import '../foo/foo.scss')
        group: 'object',
        position: 'after'
      }
    ],
    'warnOnUnassignedImports': true
  }
]
```

**Comment by vileen** [comment]
> 

`
              {
                pattern: '{.,..}/**/*.css', // same & outside directories (e.g. import '../foo/foo.scss')
                group: 'object',
                position: 'after',
              },
`
That should work

**Comment by Anshshah-1609** [comment]
same for me doesn't work this type of rules
'import/order': [
      'error',
      {
        pathGroups: [
          {
            pattern: '~/**',
            group: 'external',
            position: 'before',
          },
          {
            pattern: '~/src/logger/**',
            group: 'internal',
          },
          {
            pattern: '~/src/config/**',
            group: 'internal',
          },
          {
            pattern: '~/src/constants/**',
            group: 'internal',
          },
        ],
        groups: ['builtin', 'external', 'internal'],
      },
    ],

**Comment by Space1nvader** [comment]
in one of the answers on stackoverflow it was written:

"After some research, it seems that eslint community does not want to add auto-sorting for [unassigned imports](https://github.com/import-js/eslint-plugin-import/issues/1639#issuecomment-580862011) due to the fact, that they might depend on import order. That is the reason, why your snippet of pathGroups does not work"

original: 
https://stackoverflow.com/questions/69814021/eslint-import-order-rule-does-not-work-with-scss-files
